Dalamud/Game/Gui/ContextMenu/ContextMenuGui.cs

```diff
@@ -0,0 +1,501 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using Dalamud.Game.Text;
+using Dalamud.Game.Text.SeStringHandling;
+using Dalamud.Hooking;
+using Dalamud.IoC;
+using Dalamud.IoC.Internal;
+using Dalamud.Logging.Internal;
+using Dalamud.Memory;
+using Dalamud.Plugin.Services;
+using FFXIVClientStructs.FFXIV.Client.System.Memory;
+using FFXIVClientStructs.FFXIV.Client.UI;
+using FFXIVClientStructs.FFXIV.Client.UI.Agent;
+using FFXIVClientStructs.FFXIV.Component.GUI;
+using FFXIVClientStructs.Interop;
+
+using ValueType = FFXIVClientStructs.FFXIV.Component.GUI.ValueType;
+
+namespace Dalamud.Game.Gui.ContextMenu;
+
+/// <summary>
+/// This class handles interacting with the game's (right-click) context menu.
+/// </summary>
+[InterfaceVersion("1.0")]
+[ServiceManager.BlockingEarlyLoadedService]
```

Comment:
Use `EarlyLoadingService`. This service does not need to block the game from loading to initialize "better".